### PR TITLE
added stooge sort 

### LINKED ISSUE #148: add Stooge Sort
**Stooge Sort** is a recursive and inefficient sorting algorithm. It works by recursively sorting elements in a peculiar manner, following these steps:

1. If the first element is larger than the last element, they are swapped.
2. If the array has more than two elements, it recursively:
   - Sorts the first two-thirds of the array.
   - Sorts the last two-thirds of the array.
   - Sorts the first two-thirds of the array again.

### Time Complexity:
- **Worst-case time complexity**: O(n^(log 3 / log 1.5)) ≈ O(n^2.7095), making it highly inefficient compared to most other sorting algorithms.

Stooge Sort is more of a theoretical or humorous algorithm, demonstrating an impractical method for sorting arrays. It is rarely used in practical applications due to its poor performance.

### PR DESCRIPTION
**Pull request by ananyag309**
Fixes #148 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/algorithms/sorting-algorithms/StoogeSort.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Stooge sort is a **recursive sorting algorithm** that operates by comparing the first and last elements of a sequence and swapping them if necessary. It then recursively applies the same procedure to overlapping subarrays. While Stooge sort is notable for its academic interest, it is inefficient for practical use due to its high time complexity.
+
+### Characteristics:
+
+- **Recursive Algorithm**:
+  - Stooge sort uses a recursive approach to sort elements. It first swaps elements if needed, then recursively sorts the first two-thirds and the last two-thirds of the array.
+
+- **Inefficient Time Complexity**:
+  - Despite its simplicity, Stooge sort is inefficient with a time complexity of **O(n^2.7095)**, making it one of the least efficient sorting algorithms.
```
Comment (ajay-dhangar) [inline docs/algorithms/sorting-algorithms/StoogeSort.md]: replace `**O(n^2.7095)**` with `$O(n^{2.7095})$`

File: docs/algorithms/sorting-algorithms/StoogeSort.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+- **Rarely Used in Practice**:
+  - Due to its poor performance, Stooge sort is rarely used in real-world applications, but it remains a useful algorithm for theoretical study.
+
+### Time Complexity:
+
+- **Best Case: O(n^2.7095)**
```
Comment (ajay-dhangar) [inline docs/algorithms/sorting-algorithms/StoogeSort.md]: replace `O(n^2.7095)` with `$O(n^{2.7095})$`

File: docs/algorithms/sorting-algorithms/StoogeSort.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+- **Rarely Used in Practice**:
+  - Due to its poor performance, Stooge sort is rarely used in real-world applications, but it remains a useful algorithm for theoretical study.
+
+### Time Complexity:
+
+- **Best Case: O(n^2.7095)**  
+  Even in the best case, Stooge sort requires recursive calls that make its performance slower than more efficient algorithms like merge sort or quicksort.
+
+- **Average Case: O(n^2.7095)**
```
Comment (ajay-dhangar) [inline docs/algorithms/sorting-algorithms/StoogeSort.md]: reply `O(n^2.7095)` with `$O(n^{2.7095})$`

File: docs/algorithms/sorting-algorithms/StoogeSort.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+### Time Complexity:
+
+- **Best Case: O(n^2.7095)**  
+  Even in the best case, Stooge sort requires recursive calls that make its performance slower than more efficient algorithms like merge sort or quicksort.
+
+- **Average Case: O(n^2.7095)**  
+  The time complexity remains $O(n^{2.7095})$ across average cases due to the recursion.
+
+- **Worst Case: O(n^2.7095)**
```
Comment (ajay-dhangar) [inline docs/algorithms/sorting-algorithms/StoogeSort.md]: reply `O(n^2.7095)` with `$O(n^{2.7095})$`

File: docs/algorithms/sorting-algorithms/StoogeSort.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+- **Average Case: O(n^2.7095)**  
+  The time complexity remains $O(n^{2.7095})$ across average cases due to the recursion.
+
+- **Worst Case: O(n^2.7095)**  
+  Stooge sort's worst-case performance is similar to its average and best-case complexities.
+
+### Space Complexity:
+
+- **Space Complexity: O(n)**
```
Comment (ajay-dhangar) [inline docs/algorithms/sorting-algorithms/StoogeSort.md]: reply `O(n)` with `$O(n)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by ananyag309** [comment]
@ajay-dhangar please check

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**